### DISCUSSION TITLE
BoidCMS suggestions and the future

### DISCUSSION THREAD

**Discussion by ctuxboy** [BoidCMS/BoidCMS#38] [Ideas]
After following BoidCMS more then a year, and try the demo, installed BoidCMS for the first time.
It looks really good, and works smoother then WonderCMS. I like the small footprint from WCMS, but the UI/UX is disappointing.

BoidCMS but seems promising.
I notice that there have been no updates for a long time. I'm wonder if the project is still actively maintained, as well as if there will be updates in the near future?
I hope that there will be plugins and themes developed, which will offer the project many more possibilities.

May I be so free to propose some suggestions, what I personally miss?
- Search function (plugin?)
- Image gallery (plugin?)
- Contact form (plugin?)
- Possibility for the user customize own Actions/Tags/Blocks in themes.
- **Dashboard-page:** Show simple statistics
- **Update pages-page:** better to show a list of pages instead of a dropdown, especially for the blog pages (UX)
- **Media-page:** file name, 'view' and 'delete' next to each other (UX)
- UI/UX: Work more with inline elements on the admin pages, now there is a lot below each other elements
- last but not least keep it small ;)

I hope you find the motivation to continue working on this beautiful project, I think that the above points and more, will give BoidCMS more flexibility and more users will find their way to this beautiful project.

PS: i like the way you build the nav-function. It give the user/developer the possibility build his own nav with his own html-tags. This is very tricky by a lot of cms-systems.

Christophe

**Comment by sysadev** [comment]
Hey Christophe,

Thanks for taking the time to give this feedback on BoidCMS. I really appreciate it! I totally get your concerns about updates and future plans.

BoidCMS is an active project, and I'm committed to ongoing development and improvement. I'm currently working on a new plugin called **Modules**, which will streamline the process of installing themes and plugins, and I'm excited to share more details soon.

Your suggestions for a search function, image gallery, contact form, and more customization options are awesome. I'll definitely take them into account as I plan for the future.

Regarding your suggestion about custom actions, I just wanted to let you know that it's already possible for users to use custom actions in themes. Is that what you meant by "Possibility for the user customize own Actions/Tags/Blocks in themes."?

And about your suggestion for the media page, I just wanted to let you know that the file name, 'view,' and 'delete' options are already placed next to each other.

Thanks again for your support and enthusiasm. If you're interested in contributing to the development of BoidCMS, please don't hesitate to reach out.

**Reply by ctuxboy** [reply]
> Hey Christophe,
> 
> Thanks for taking the time to give this feedback on BoidCMS. I really appreciate it! I totally get your concerns about updates and future plans.
> 
> BoidCMS is an active project, and I'm committed to ongoing development and improvement. I'm currently working on a new plugin called **Modules**, which will streamline the process of installing themes and plugins, and I'm excited to share more details soon.
> 
> Your suggestions for a search function, image gallery, contact form, and more customization options are awesome. I'll definitely take them into account as I plan for the future.
> 
> Regarding your suggestion about custom actions, I just wanted to let you know that it's already possible for users to use custom actions in themes. Is that what you meant by "Possibility for the user customize own Actions/Tags/Blocks in themes."?
> 
> And about your suggestion for the media page, I just wanted to let you know that the file name, 'view,' and 'delete' options are already placed next to each other.
> 
> Thanks again for your support and enthusiasm. If you're interested in contributing to the development of BoidCMS, please don't hesitate to reach out.

@shoaiyb Thank you for your reply, so i know you actively work on the project , and this is awesome.
I build a [test-website](https://dev.gewoonsimpel.be/) with WCMS, but i will try also build the same website with your awesome cms. This evening i reading (again) the BoidCMS docs, and every time read the docs, so more understand how flexible and the possibilities BoidCMS has. 

hmm, about the admin-pages, all the fields and buttons are below each other. See my screenshots...

![Screenshot 2024-10-19 19 59 08](https://github.com/user-attachments/assets/f6a03449-2c22-4c42-8bbd-bc7ae8ae8848)
![Screenshot 2024-10-19 19 58 53](https://github.com/user-attachments/assets/e089c31e-6245-4403-9deb-df7c8457ce25)

So i said, i like te small WCMS foodprint, so i can focus building my own 'small' template, but there is a lot i missing in this cms (list pages is tricky, there are no keywords in the simple blog plugin,... all required basic stuff.

My inspiration for building small websites came from this [website](https://512kb.club/).

About contributing, i'm not a pro programmer, so my skills are basic (html,php,css,js, ...) i can help myself with customize small code-snippets in (most cases) procedural php.
The more I get familiar with BoidCMS, there might be an opportunity to consider experimenting with building a plugin/theme myself and sharing it with the community.

I am following your project with great interest, and will definitely experiment with it, and possibly try develope a project with my own template.

Last question: what i understand from the docs, that it is possible customize the admin-pages? If yes, and you have time can you explain how list the pages (possibly with the published date next to it) instead of a dropdown  on the update-page, by example:
tip: Maybe you can make as a tutorial for the docs section?
-------------------------------UPDATE PAGES------------------------------------------------
- page title (post)       published       template-name         20/09/24   edit
- page title (page)      published       template-name        23/09/24    edit
- page title (post)       published       template-name        30/09/24    edit
- page title (post)       unpublished   template-name        14/10/24   edit
- ...
-----------------------------------------------------------------------------------------------------

(Sorry for my bad english)

**Reply by sysadev** [reply]
First of all, so happy that you liked BoidCMS. And again, what makes BoidCMS standout is **extensibility**.

From the screenshots, are you saying that the buttons and fields are too close?
It is also possible to modify or make new admin pages, I even made a short example on how one could make a new page (https://boidcms.github.io/#/developer/panel?id=example-code) but if that isn't very clear then I'll see what I can do about the tutorial, possibly one with a demo.

**Reply by ctuxboy** [reply]
Thank you for your answer :)
After reading the docs again 🙈, found indeed a template how change the admin. 
Sometimes i need a good example tutorial for understanding better how things works. 
Okay i'm a newbie and not so 'smart', so reading the docs several times, then i understand better and found also new things 😋
Last days experimenting a lot with BoidCMS.
